### PR TITLE
Re-enable multizone fully supported

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
# Description
Bump egoscale with the new List zones mapping feature https://github.com/exoscale/egoscale/pull/616
Push the Multizone to the finish-line with the new egoscale version 

## Checklist
(For exoscale contributors)

* [ ] Changelog updated (under *Unreleased* block)
* [x] Integration tests OK

### CODE REVIEW

File: driver/controller.go [hunk 1]
```diff
@@ -94,30 +92,57 @@ func newControllerService(client *v3.Client, nodeMeta *nodeMetadata) controllerS
...
-		if v.Name == req.Name {
-			return &csi.CreateVolumeResponse{
-				Volume: &csi.Volume{
-					VolumeId: exoscaleID(d.zoneName, v.ID),
-					// API reply in bytes then send it without conversion
-					CapacityBytes:      v.Size,
-					AccessibleTopology: newZoneTopology(d.zoneName),
-				},
-			}, nil
+	// Fetch all zones volumes
```
Comment (pierre-emmanuelJ) [inline driver/controller.go:102]: Maybe only for the `createVolume` (not for the list..etc), only use the zone returned by the`AccessibilityRequirements` topology

File: driver/controller.go [hunk 2]
```diff
@@ -94,30 +92,57 @@ func newControllerService(client *v3.Client, nodeMeta *nodeMetadata) controllerS
...
+			if v.Name == req.Name {
+				return &csi.CreateVolumeResponse{
+					Volume: &csi.Volume{
+						VolumeId: exoscaleID(zone.Name, v.ID),
+						// API reply in bytes then send it without conversion
+						CapacityBytes:      v.Size,
+						AccessibleTopology: newZoneTopology(zone.Name),
+					},
+				}, nil
+			}
```
Comment (sauterp) [inline driver/controller.go:128]: I think this change will now prevent users from creating a volume with the same name in different zones. But is this a real constraint of block storage? Do block storage volume names have to be unique across zones?
If no, then we should make this idempotent for the zone of the cluster only.

Comment (pierre-emmanuelJ) [reply driver/controller.go:128]: Maybe this question can be gone if we decide to do this:
https://github.com/exoscale/exoscale-csi-driver/pull/9/files/908ce54b376e423f416cabd33633ee64fd19e6cd#r1481087895

File: driver/controller.go [hunk 3]
```diff
@@ -196,13 +220,20 @@ func (d *controllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
...
 		return nil, err
 	}
 
-	op, err := d.client.DeleteBlockStorageVolume(ctx, volumeID)
+	endpoint, err := d.client.GetZoneAPIEndpoint(ctx, zoneName)
+	if err != nil {
+		klog.Errorf("delete volume: get zone endpoint: %v", err)
+		return nil, err
+	}
+	client := d.client.WithEndpoint(endpoint)
```
[the remark this answers is not in the document]
Comment (pierre-emmanuelJ) [reply driver/controller.go]: https://github.com/exoscale/exoscale-csi-driver/pull/9/commits/162e1104d23ca443b98004496297aa2d11062dab

**Review by sauterp — APPROVED** [review]

**Pull request merged**